gemd/entity/template/has_property_templates.py

```diff
@@ -17,7 +17,7 @@ class HasPropertyTemplates(HasDependencies):
 
     Parameters
     ----------
-    properties: List[(PropertyTemplate, BaseBounds)]
+    properties: List[(~gemd.entity.template.property_template.PropertyTemplate, BaseBounds)]
```

Comment:
This type for `properties` is quite different from the type hints on the actual parameter.

Reply:
An English language description of the type here is a list-like object, where each element of the list is one of:
* A PropertyTemplate
* A LinkByUID that maps to a PropertyTemplate (unknowable / unenforceable)
* A Tuple that contains:
    * A PropertyTemplate or a LinkByUID
    * A Bounds object

The signature lines up with that, but that's a pretty complicated thing to put into documentation.  Probably just needs an essay.

Reply:
If we're going to document the types, then we should be as accurate and precise as possible. Otherwise, there's no point.

If we can't describe the types we accept as English, then using a code like structure similar to the type hint is a reasonable move. Because while it's not as important that the documentation of inputs cover every case, that both type hints show up in the docs may cause some confusion.

But I'm more concerned with disagreements between the docstrings and type hints with properties and other values to be read. For example, `parameters` documents its return as `List[Union[ParameterTemplate, LinkByUID]]`, which is quite different from the docstring's claim of ` List[(ParameterTemplate, BaseBounds)]`. A list of elements which are either `ParameterTemplate` or `LinkByUID` objects is not the same as a list of tuples, where each tuple is one `ParameterTemplate` followed by a `BaseBounds`.

Although this raises the question of why we must include the type in the docstring and inline. It would seem to be redundant, as the generated Sphinx docs include both, and thus just introduces chances for conflict, like this.